coalib/misc/Caching.py

```diff
@@ -0,0 +1,142 @@
+import os
+import calendar
+import time
+import appdirs
+import pickle
```

Comment:
@fneu I've added a `verify_file_integrity` function that will check if all files are proper. It will delete all of them if even one of them is corrupted (doesn't make sense to have the others if they are all connected). This is done at the start of each coala execution (before the first time any pickling is done).